browser/ui/toolbar/brave_bookmark_context_menu_controller.cc

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/toolbar/brave_bookmark_context_menu_controller.h"
+
+#include <memory>
+#include <vector>
+
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/ui/bookmarks/bookmark_context_menu_controller.h"
+#include "chrome/grit/generated_resources.h"
+
+BraveBookmarkContextMenuController::BraveBookmarkContextMenuController(
+    gfx::NativeWindow parent_window,
+    BookmarkContextMenuControllerDelegate* delegate,
+    Browser* browser,
+    Profile* profile,
+    BookmarkLaunchLocation opened_from,
+    const bookmarks::BookmarkNode* parent,
+    const std::vector<const bookmarks::BookmarkNode*>& selection)
+    : BookmarkContextMenuController(parent_window,
+                                    delegate,
+                                    browser,
+                                    profile,
+                                    opened_from,
+                                    parent,
+                                    selection) {
+  AddBraveBookmarksSubmenu(profile);
+}
+
+BraveBookmarkContextMenuController::~BraveBookmarkContextMenuController() =
+    default;
+
+void BraveBookmarkContextMenuController::AddBraveBookmarksSubmenu(
+    Profile* profile) {
+  auto index = menu_model()->GetIndexOfCommandId(IDC_BOOKMARK_BAR_ALWAYS_SHOW);
+  if (!index.has_value())
+    return;
+  menu_model()->RemoveItemAt(index.value());
+  brave_bookmarks_submenu_model_ =
+      std::make_unique<BookmarkBarSubMenuModel>(profile);
+
+  menu_model()->AddSubMenuWithStringId(IDC_BRAVE_BOOKMARK_BAR_SUBMENU,
```

Comment:
don't we need to make sure this gets inserted at the same index?

Reply:
done, thanks